### DISCUSSION TITLE
Why not use arch viewModelScope in Android ViewModel

### DISCUSSION THREAD

**Discussion by osrl** [icerockdev/moko-mvvm#132] [General]
I was checking your library to see if we can use it in our new project (great work by the way). What is the reason that you don't use viewModelScope of architecture ViewModel for Android? Also, I saw that #110  you will move LiveData implementation to common. Won't this make you move away from architecture LiveData?

**Comment by svenjacobs** [comment]
I was thinking the same. Actually (pun intended 😉) this should be pretty simple:

```kotlin
expect open class ViewModel() {

    val viewModelScope: CoroutineScope

    protected fun onCleared()
}
```

and in `androidMain`:

```kotlin
import androidx.lifecycle.ViewModel as AndroidViewModel
import androidx.lifecycle.viewModelScope as androidViewModelScope

actual open class ViewModel : AndroidViewModel() {

    actual val viewModelScope: CoroutineScope
        get() = androidViewModelScope

    actual override fun onCleared() {
    }
}
```
